lib/matplotlib/backends/_macos_helpers.py

```diff
@@ -0,0 +1,10 @@
+import sys
```

Comment:
I would not even bother with testing the os and just make the module only importable on OSX (just like backend_macos.py).

Reply:
Also, I have to wrap at some level because this will wind up in other backend code that’s platform-agnostic, like the Qt5 part. I’m being extra-paranoid, though, I actually check twice. Maybe only one level of paranoia?

How should we resolve this?

Reply:
Just wrap in once before importing it and call it a day.

Reply:
I could get rid of the _macos_helpers.py entirely if we just make sure to wrap imports of _macosx with platform checks, for instance... It's sort of a style and paranoia level choice. Lots of ways to skin the cat, just not sure how you'd like it done.

Reply:
I personally don't like duplicated checks, but won't insist on having absolutely nothing done twice.

Reply:
Kind of like this?

```
import sys

if sys.platform == "darwin":
    from ._macos_helpers import set_icon as set_mac_icon
else:
    def set_mac_icon(path):
        pass
```

Then in the Qt5 backend, I can do:

```
from matplotlib.backends import _macos_helpers

...

_macos_helpers.set_mac_icon(...)
```

Reply:
I would not have _macos_helpers at all (one less module, yay), and in qt5 I would have
```
if sys.platform == "darwin":
    import _macos; _macos.set_mac_icon(...)
```

Reply:
I'm good with that. Where would I put the tests? There don't seem to be tests for macOS. Or do I still need them?

Reply:
I wouldn't bother with these tests.  I mean, they're not even actually testing the real functionality, but just that the correct error is thrown when a private function is called with stuff that it's never going to be called with...

Reply:
Well, mostly I just wanted to make sure the code wouldn’t crash when given bad input... I could imagine somebody moves the matplotlib.pdf in the repo, and then all of their tests crash hard...

On the other hand, if the code did crash, these tests wouldn’t really help with narrowing down the issue...

Your call. Test or don’t test?

Reply:
Sure, you can put one, I guess in test__macosx.

Reply:
Okay, done. Latest push has the test. Another review?